dashboard/test/models/user_test.rb

```diff
@@ -7,6 +7,28 @@ class UserTest < ActiveSupport::TestCase
   include ProjectsTestUtils
   self.use_transactional_test_case = true
 
+  class UsStateCodeTest < ActiveSupport::TestCase
+    test 'returns student us_state if present' do
+      student = create(:student, :in_colorado)
+      assert_equal 'CO', student.us_state_code
+    end
+
+    test 'returns nil if student us_state is "??"' do
+      student = create(:student, :unknown_us_region)
+      assert_nil student.us_state_code
+    end
+
+    test 'returns teacher school US state code' do
+      teacher = create(:teacher, school_info: create(:school_info, state: 'ny'))
+      assert_equal 'NY', teacher.us_state_code
+    end
+
+    test 'returns teacher school US state code when state is name' do
+      teacher = create(:teacher, school_info: create(:school_info, state: 'washington dc'))
+      assert_equal 'DC', teacher.us_state_code
+    end
```

Comment:
Maybe we should cover the case where the `school_info` does not have a state recorded or has an unknown state ... maybe it is a school in Canada, Europe, etc. I think you handle that case in your implementation.

Reply:
Basically, cover the case where the teacher's state would be `nil` I think.

Reply:
It stores only U.S. states

Reply:
https://github.com/code-dot-org/code-dot-org/pull/58688/commits/e187410d8cb2af0c28f764763d420bca925c9955